src/include/sof/audio/audio_stream.h

```diff
@@ -192,6 +228,82 @@ static inline void *audio_stream_wrap(const struct audio_stream *buffer,
 	return ptr;
 }
 
+/**
+ * Calculates available data in bytes, handling underrun_permitted behaviour
+ * @param stream Stream pointer
+ * @return amount of data available for processing in bytes
+ */
+static inline uint32_t
+audio_stream_get_avail_bytes(const struct audio_stream *stream)
+{
+	if (!stream->underrun_permitted)
```

Comment:
inline comment here

Reply:
I would also deal with the underrun permitted here (next to it's conditional logic) than as the default return case.

Reply:
done